components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTNotificationStrategyTest.java

```diff
@@ -0,0 +1,176 @@
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.push.notification.NotificationContext;
+import org.wso2.carbon.device.mgt.common.push.notification.PushNotificationConfig;
+import org.wso2.carbon.device.mgt.common.push.notification.PushNotificationExecutionFailedException;
+import org.wso2.carbon.device.mgt.core.operation.mgt.PolicyOperation;
+import org.wso2.carbon.device.mgt.core.operation.mgt.ProfileOperation;
+import org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt.internal.MQTTDataHolder;
+import org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt.internal.util.MQTTAdapterConstants;
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+import org.wso2.carbon.event.output.adapter.core.internal.CarbonOutputEventAdapterService;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.device.mgt.common.operation.mgt.Operation;
+import static org.wso2.carbon.device.mgt.core.operation.mgt.PolicyOperation.POLICY_OPERATION_CODE;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/*
+    Unit tests for MQTTNotificationStrategy class
+ */
+public class MQTTNotificationStrategyTest {
+    private MQTTNotificationStrategy mqttNotificationStrategy;
+    private CarbonOutputEventAdapterService carbonOutputEventAdapterService;
+    private static final String ADAPTER_NAME = "SampleMqttAdapterName";
+    private static final String BROKER_URL = "SampleBrokerUrl";
+    private PushNotificationConfig pushNotificationConfig;
+    private static final String MQTT_ADAPTER_TOPIC = "mqtt.adapter.topic";
+    private DeviceIdentifier deviceIdentifier;
+    private Operation operation;
+    private Map<String, String> propertiesMap;
+    private NotificationContext notificationContext;
+
+    @BeforeClass
+    public void init() throws NoSuchFieldException, IllegalAccessException, IOException, RegistryException,
+            OutputEventAdapterException {
+        initializeCarbonContext();
+        mqttNotificationStrategy = Mockito.mock(MQTTNotificationStrategy.class, Mockito.CALLS_REAL_METHODS);
+        carbonOutputEventAdapterService = Mockito.mock(CarbonOutputEventAdapterService.class,
+                Mockito.CALLS_REAL_METHODS);
+        Mockito.doReturn(true).when(carbonOutputEventAdapterService).isPolled(Mockito.any());
+        Mockito.doNothing().when(carbonOutputEventAdapterService).publish(Mockito.any(), Mockito.any(), Mockito.any());
+        Mockito.doNothing().when(carbonOutputEventAdapterService).destroy(ADAPTER_NAME);
+    }
+
+    private void initializeCarbonContext() throws IOException, RegistryException {
+        if (System.getProperty("carbon.home") == null) {
```

Comment:
This method is duplicated  with MQTTBasedPushNotificationProviderTest#initializeCarbonContext() can't we introduce a util for this ?